src/fonts/castletexturefontdata.pas

```diff
@@ -313,9 +315,22 @@ constructor TTextureFontData.Create(const URL: string;
   FontMgr.Resolution := 0;
   FileName := URIToFilenameSafe(URL);
   if FileName = '' then
-    raise Exception.CreateFmt('Cannot read font from URL "%s". Note that right now only local file URLs are supported', [URL]);
+  begin
+    Cache := Download(URL);
+    try
+      FileName := ApplicationConfig('cache_' + ExtractURIName(URL));
+      StreamSaveToFile(Cache, FileName);
+      FileName := URIDeleteProtocol(FileName);
+      IsCachedFile := true;
+    finally
+      Cache.Free;
+    end;
+  end;
   FontId := FontMgr.RequestFont(FileName);
 
+  if IsCachedFile then
+    DeleteFile(FileName);
+
```

Comment:
Use `CheckDeleteFile(FileName, true)` instead `DeleteFile(FileName)`.

The `CheckDeleteFile` is a trivial utility in CastleFilesUtils, it calls `DeleteFile` and checks the result, and makes a warning if it failed. So a failure in deletion will not cause the program crash -- but it will be visible in log file, and may help spot eventual errors.